superset/connectors/sqla/models.py

```diff
@@ -721,43 +721,46 @@ def get_sqla_query(  # sqla
             if not all([flt.get(s) for s in ["col", "op"]]):
                 continue
             col = flt["col"]
+
+            if col not in cols:
+                raise Exception(_(f"Column '{col}' is not valid"))
```

Comment:
maybe we could update the language to be a little more precise here? For example `Column '{col}' does not exist`  / `... not found`?

Reply:
I think the wording here can be difficult as it's in reference to Superset's data model as opposed to the schema of the underlying table. I think "does not exist" is more precise/clearer than "is not valid" though.